File: 2245-destroying-asteroids/destroying-asteroids.java
class Solution {
    public boolean asteroidsDestroyed(int mass, int[] asteroids) {
        int n=asteroids.length;

        long ms=mass;
        Arrays.sort(asteroids);
        for(int a: asteroids){
            if(ms<a)
                return false;

            ms += a;
        }

        return true;
    }
}

File: Medium
Medium/2245-destroying-asteroids/destroying-asteroids.cpp
#define ll long long int

class Solution{
public:
    bool asteroidsDestroyed(int m, vector<int> &a){
        int n=a.size();

        ll ms=m;
        sort(a.begin(), a.end());
        for(int i=0; i<n; i++){
            if(ms<a[i])
                return false;

            ms += a[i];
        }

        return true;

    }
};